### ISSUE TITLE
mdbx_env_open: cannot allocate memory on 2TiB of ram numa machine and with 900Gib free.

### ISSUE THREAD

**Issue by ytrezq** [ledgerwatch/erigon#5335]
I used the latest devel branch compiled from yesterday. `/bin/erigon --experimental.overlay --nat upnp --datadir=/tmp/ETherErigon/ --watch-the-burn --vmdebug --blockDownloaderWindow=4194304 --v5disc`
log is too Big for GitHub, so I put it on [pastebin](https://pastebin.com/K3t30x8J).

**Comment by ytrezq** [comment]
System was reporting 14.9Tib allocated along 227Gib memory usage by Erigon.

**Comment by AskAlexSharov** [comment]
maybe duplicate of https://github.com/ledgerwatch/erigon/issues/4871#issuecomment-1229275707

**Comment by ytrezq** [comment]
@AskAlexSharov adressing is 48bits physical and virtual in my case. That s lot of memory. Beside using ibm power or sparc, there isn t much I can do.

**Comment by AskAlexSharov** [comment]
I see another error before "allocate":
```
EROR[09-11|21:01:35.049] Staged Sync                              err="append with gap blockNum=15000000, but current heigh=1, trace: [stageloop.go:136 panic.go:890 txNums.go:34 stage_bodies.go:210 default_stages.go:59 sync.go:352 sync.go:255 stageloop.go:170 stageloop.go:91 asm_amd64.s:1594]"
```
so, we need to fix this one. 
do you see it when restart erigon?

**Comment by ytrezq** [comment]
@AskAlexSharov no but erigon is now looping over the same block.

**Comment by AskAlexSharov** [comment]
`append with gap` - ah, i know what is it. will fix now

**Comment by AskAlexSharov** [comment]
fixed by https://github.com/ledgerwatch/erigon/pull/5340

**Comment by rotsen11** [comment]
😝

Obtener Outlook para iOS<https://aka.ms/o0ukef>
________________________________
De: Max Revitt ***@***.***>
Enviado: Monday, September 19, 2022 8:46:43 AM
Para: ledgerwatch/erigon ***@***.***>
CC: Subscribed ***@***.***>
Asunto: Re: [ledgerwatch/erigon] mdbx_env_open: cannot allocate memory on 2TiB of ram numa machine and with 900Gib free. (Issue #5335)


Closed #5335<https://github.com/ledgerwatch/erigon/issues/5335> as completed.
